File: leetcode/201-1000/leetcode205.cpp
#include <algorithm>
#include <cstring>
#include <iostream>
#include <map>
#include <string>
using namespace std;
class Solution {
public:
    bool isIsomorphic(string s, string t)
    {
        map<char, char> m;
        int len = s.length();
        if (t.length() != len)
            return false;
        for (int i = 0; i < len; i++) {
            if (!m[s[i]]) {
                m[s[i]] = t[i];
            } else if (m[s[i]] != t[i]) {
                return false;
            }
        }
        m.clear();
        for (int i = 0; i < len; i++) {
            if (!m[t[i]]) {
                m[t[i]] = s[i];
            } else if (m[t[i]] != s[i]) {
                return false;
            }
        }
        return true;
    }
};
int main(void)
{
    return 0;
}

File: leetcode/100-999/leetcode205.py
class Solution:
    def isIsomorphic(self, s: str, t: str) -> bool:
        dic = {}
        if len(s) != len(t):
            return False
        for i in range(len(s)):
            if not dic.get(s[i]) and t[i] not in dic.values():
                dic[s[i]] = t[i]
            elif dic.get(s[i]):
                if t[i] != dic[s[i]]:
                    return False
            else:
                return False
        return True